### ISSUE TITLE
[4.4.0] Joomla! uses wrong update package when re-installing Joomla! core files

### ISSUE THREAD

**Issue by maggus** [joomla/joomla-cms#42155]
### Steps to reproduce the issue

After updating from Joomla! 4.3.4 to 4.4.0 and when trying to re-install the Joomla! Core Files it updates to 5.0.0 without any prior notification.

This happens on different websites and I feel like this is a serious bug! I am using the Joomla! "Standard" Update Channel which is supposed to only provide updates for 4.x Major version, right?

**Comment by ToxicTeemanIsWrong** [comment]
Confirmed - see https://github.com/joomla/joomla-cms/issues/42154#issuecomment-1767270393

**Comment by richard67** [comment]
The update channel in the Joomla Update component, was that set to „Default“? Or to „Next“? Not that this would justify the error, only like to know for better diagnostics and investigation.

**Comment by richard67** [comment]
The issue has been temporarily fixed with a workaround on our update site. See https://github.com/joomla/joomla-cms/issues/42154#issuecomment-1767333179 for details.

**Comment by maggus** [comment]
The Joomla! Update Channel was intentionally set to "Default". But it did not respect the bounds of the Joomla! 4.x version.

Thank you for fixing it!

**Comment by brianteeman** [comment]
For reference why were you trying to reinstall the update so soon

**Comment by richard67** [comment]
The issue here has been solved with https://github.com/joomla/update.joomla.org/pull/335 for the update sites. I have tested with 4.4.0 and 5.0.0 that the reinstall core files button is shown when it should be shown (no update found after the check button was used) and that it results in the right core files for the right version and in case of 4.4.0 that the update to 5.0.0 is found when the "Next" update channel is selected and that that upüdate from 4.0.0 to 5.0.0 still works.

Thanks a lot to @HLeithner and @bembelimen for the fix.
